src/check_mailgun_esp_blocks.py

```
@@ -0,0 +1,308 @@
+
+            dimensions_list = item.get("dimensions", [])
+            metrics = item.get("metrics", {})
+
+            # Ensure metrics is a dictionary
+            if not isinstance(metrics, dict):
+                metrics = {}
+
+            # Extract domain and provider from dimensions list
+            domain = "Unknown"
+            provider = "Unknown"
+
+            if isinstance(dimensions_list, list):
+                for dim in dimensions_list:
+                    if isinstance(dim, dict):
+                        dim_name = dim.get("dimension", "")
+                        dim_value = dim.get("value", "")
+                        if dim_name == "domain":
+                            domain = dim_value
+                        elif dim_name == "recipient_provider":
+                            provider = dim_value if dim_value else "unknown"
+
+            permanent_count = metrics.get("permanent_failed_esp_block_count", 0)
+            temporary_count = metrics.get("temporary_failed_esp_block_count", 0)
+            total_blocks = permanent_count + temporary_count
+
+            # Only include domains with blocks
+            if total_blocks > 0:
+                domain_issues.append(
+                    {
+                        "domain": domain,
+                        "provider": provider,
+                        "permanent": permanent_count,
+                        "temporary": temporary_count,
+                        "total": total_blocks,
+                    }
+                )
+
+            total_permanent += permanent_count
+            total_temporary += temporary_count
+
+        # Sort by total blocks descending
+        domain_issues.sort(key=lambda x: x["total"], reverse=True)
+
+        return total_permanent, total_temporary, domain_issues
+
+    except (KeyError, TypeError, AttributeError) as e:
+        raise Exception(
+            f"Error processing response data: {e}. Response type: {type(response_data)}"
+        )
```

Same here. I would rather say we attempt to implement everything correct and if not let it raise the error with all it's details some sysadmin/developer has to check anyways if it does not work - it's not customer facing.